2/Date.rb

```diff
@@ -0,0 +1,21 @@
+=begin
+Заданы три числа, которые обозначают число, месяц, год (запрашиваем у пользователя). 
+Найти порядковый номер даты, начиная отсчет с начала года. Учесть, что год может быть високосным. 
+(Запрещено использовать встроенные в ruby методы для этого вроде Date#yday или Date#leap?) 
+Алгоритм опредления високосного года: www.adm.yar.ru
+=end
+
+puts "Введите число:"
+number = gets.to_i
+puts "Введите месяц:"
+month = gets.to_i
+puts "Введите год:"
+year = gets.to_i
+
+leap_year = year % 4 == 0 && (year % 100 != 0 || year % 400 == 0) ? true : false
+
+days_in_year = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
+days_in_year[1] = 29 if leap_year == true
```

Comment:
```suggestion
days_in_year[1] = 29 if leap_year
```
`leap_year` уже boolean